libr/bin/p/bin_xtr_fatmach0.c

```diff
@@ -13,13 +13,19 @@ static RBinXtrData * oneshot(RBin *bin, const ut8 *buf, ut64 size, int idx);
 static RList * oneshotall(RBin *bin, const ut8 *buf, ut64 size );
 static int free_xtr (void *xtr_obj) ;
 
-static bool checkHeader(const ut8 *h, ut64 sz) {
+static bool checkHeader(RBuffer *b) {
 	ut8 buf[4];
-	if (sz >= 0x300 && !memcmp (h, "\xca\xfe\xba\xbe", 4)) {
+	const ut64 sz = r_buf_size (b);
+	r_buf_read_at (b, 0, buf, 4);
+	if (sz >= 0x300 && !memcmp (buf, "\xca\xfe\xba\xbe", 4)) {
 		// XXX assuming BE
-		ut64 off = (ut64)r_read_at_be32 (h, 4 * sizeof (ut32));
+		// ut64 off = (ut64)r_read_at_be32 (h, 4 * sizeof (ut32));
+		// (ut64)r_read_at_be32 (h, 4 * sizeof (ut32));
+		r_buf_seek (b, 4 * sizeof (ut32), 0);
+		ut64 off = (ut64) r_buf_read_be32 (b);
```

Comment:
no need to do seek+r_buf_read, you can use `r_buf_read_be32_at`.

Reply:
done